Replace FINAL_11 with final.

chrisdembia opened the pull request:
The changes to OpenSim/Wrapping/SWIGSimTK/common.h should be discussed. See some discussion over at simbody [here](https://github.com/simbody/simbody/pull/302/files#diff-ba43f68612864196b99e9cd542e95b73L239). This common.h file in simbody was just updated by @sherm1.

chrisdembia commented:
I made a change that needs discussion. It seemed like most of the virtual methods in SimpleProperty and ObjectProperty were declared final. So I decided to make these final classes instead of declaring all the virtual methods as final. Is this okay?

sherm1 commented:
>  It seemed like most of the virtual methods in SimpleProperty and ObjectProperty were declared final. So I decided to make these final classes instead of declaring all the virtual methods as final. Is this okay?

That's interesting. I'm not sure why I wanted those `final` in the first place but I think it was because there is no reasonable way to further override those methods. It isn't as clear that the whole class should be final, though. It makes sense to derive from a class even without overriding any of its methods -- it is very common to do it to provide defaults on construction, or just to add some more stuff to the class. 

So making a class final is not the same thing as making all its overridden methods final -- class final is *much* more restrictive. I understand there was big argument about it among the standards committee folks and many people are uncomfortable with the whole idea. Anyway in this case I wouldn't use it because I don't think the intent was to prevent derivation from the class, just to prevent overriding of methods that can't reasonably be overridden again.

chrisdembia commented:
> Anyway in this case I wouldn't use it because I don't think the intent was to prevent derivation from the class, just to prevent overriding of methods that can't reasonably be overridden again.

I reverted that change. Completely makes sense now that class final vs final on all virtual methods is different.

I also removed a handful of unnecessary "virtual" keywords from the old property system.

chrisdembia commented:
Ready for review.

sherm1 commented:
Looks great -- it's especially nice to see all those spurious `virtual` keywords gone! Ready to merge IMO. Needs one more reviewer I think.

chrisdembia commented:
Free granola bar to anyone else willing to review :)

chrisdembia commented:
bump

tkuchida commented on OpenSim/Simulation/Model/ActuatorForceProbe.h:
```diff
@@ -145,4 +145,4 @@ namespace OpenSim {
 
-#endif // OPENSIM_ACTUATOR_FORCE_PROBE_H_
\ No newline at end of file
+#endif // OPENSIM_ACTUATOR_FORCE_PROBE_H_
```
Bonus marks :white_check_mark:

chrisdembia replied:
my editor did this automatically I think :smile:

tkuchida commented:
For what it's worth (which admittedly isn't much), I didn't find anything awry (except for [this](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Actuators/Thelen2003Muscle.h#L240)). `FINAL_11` was changed to `final`, and I confirmed that `virtual` doesn't appear on the same line as `override`, as per Sherm's comment.

chrisdembia commented:
That's two reviews. Would anyone like to merge?

chrisdembia commented:
Thank you!
